jsapp/js/utils.es6

```diff
@@ -49,6 +49,46 @@ export function surveyToValidJson(survey) {
   return JSON.stringify(survey.toFlatJSON());
 }
 
+// TRANSLATIONS HACK (Part 2/2):
+// this function reverses nullifying default language - use it just before saving
+export function unnullifyTranslations(surveyDataJSON, assetContent) {
+  let surveyData = JSON.parse(surveyDataJSON);
+
+  let translatedProp;
+  if (assetContent.translated) {
+     translatedProp = assetContent.translated[0];
```

Comment:
Please see my comments on part 1 of the translations hack. Could this nullify / denullify cycle be split out into utility functions and unit tested?

What I've used for reviewing is this sample library question:
```json
{
    "settings": {},
    "translations": [
        "a",
        "b"
    ],
    "choices": [
        {
            "$kuid": "Wz8TGjmGl",
            "$autovalue": "choice_name",
            "list_name": "thingy",
            "name": "choice_name",
            "label": [
                "a lang c label",
                "b lang c label"
            ]
        }
    ],
    "survey": [
        {
            "select_from_list_name": "thingy",
            "name": "q_name",
            "hint": [
                "a lang hint",
                "b lang hint"
            ],
            "$autoname": "q_name",
            "constraint_message": [
                "a lang constr msg",
                "b lang constr msg"
            ],
            "label": [
                "a lang q label",
                "b lang q label"
            ],
            "required_message": [
                "a lang req",
                "b lang req"
            ],
            "$kuid": "h0Mv98wlJ",
            "type": "select_one"
        }
    ],
    "translated": [
        "constraint_message",
        "hint",
        "label",
        "required_message"
    ],
    "schema": "1"
}
```

Unfortunately, it gets all shuffled up simply by being opened and re-saved in the form editor. The difference between before and after (with keys sorted):

```diff
--- before_s.json	2018-09-09 23:40:53.974885894 -0400
+++ after_s.json	2018-09-09 23:41:00.246885672 -0400
@@ -4,6 +4,7 @@
             "$autovalue": "choice_name",
             "$kuid": "Wz8TGjmGl",
             "label": [
+                null,
                 "a lang c label",
                 "b lang c label"
             ],
@@ -19,18 +20,23 @@
             "$kuid": "h0Mv98wlJ",
             "constraint_message": [
                 "a lang constr msg",
+                null,
                 "b lang constr msg"
             ],
             "hint": [
+                null,
                 "a lang hint",
                 "b lang hint"
             ],
             "label": [
+                null,
                 "a lang q label",
                 "b lang q label"
             ],
             "name": "q_name",
+            "required": false,
             "required_message": [
+                null,
                 "a lang req",
                 "b lang req"
             ],
@@ -46,6 +52,7 @@
     ],
     "translations": [
         "a",
+        null,
         "b"
     ]
 }
```
Full before and after JSON: 
[before and after (library json).zip](https://github.com/kobotoolbox/kpi/files/2365056/before.and.after.library.json.zip)